crates/rpc/rpc/src/eth/api/server.rs

```diff
@@ -44,7 +44,11 @@ where
 
     /// Handler for: `eth_coinbase`
     async fn author(&self) -> Result<Address> {
-        Err(internal_rpc_err("unimplemented"))
+        let (_cfg_env, block_env, _block_id) =
+            // TODO: swap out latest block with pending block when that gets implemented
+            // self.evm_env_at(BlockId::Number(BlockNumberOrTag::Pending)).await?;
+            self.evm_env_at(BlockId::Number(BlockNumberOrTag::Latest)).await?;
+        Ok(block_env.coinbase)
```

Comment:
eth_coinbase is only relevant in block building and is a per node setting so we can't use the latest state.

I'm not sure how this would be implemented,

ref previous attempt to implement this: 
https://github.com/paradigmxyz/reth/pull/1597#discussion_r1123170398

this handler is usually also disabled, eg:
```
cast rpc eth_coinbase --rpc-url "https://eth-mainnet.alchemyapi.io/v2/Lc7oIGYeL_QvInzI0Wiu_pOZZDEKBrdf"
```